src/CLI/actioner/sonic-cli-lldp.py

```diff
@@ -21,79 +21,63 @@
 import time
 import json
 import ast
-import openconfig_lldp_client
+import cli_client as cc
+import collections
 from rpipe_utils import pipestr
-from openconfig_lldp_client.rest import ApiException
 from scripts.render_cli import show_cli_output
 
-
-import urllib3
-urllib3.disable_warnings()
-plugins = dict()
-
-
-def register(func):
-    """Register sdk client method as a plug-in"""
-    plugins[func.__name__] = func
-    return func
-
-def call_method(name, args):
-    method = plugins[name]
-    return method(args)
-
-def generate_body(func, args):
+def invoke_api(func, args):
+    api = cc.ApiClient()
     body = None
-    if func.__name__ == 'get_openconfig_lldp_lldp_interfaces':
-        keypath = []
-    elif func.__name__ == 'get_openconfig_lldp_lldp_interfaces_interface':
-        keypath = [args[1]]
+
+    if func == 'get_openconfig_lldp_lldp_interfaces':
+       path = cc.Path('/restconf/data/openconfig-lldp:lldp/interfaces')
+       return api.get(path)
+    elif func == 'get_openconfig_lldp_lldp_interfaces_interface':
+       path = cc.Path('/restconf/data/openconfig-lldp:lldp/interfaces/interface={name}', name=args[1])
+       return api.get(path)
     else:
        body = {}
 
-    return keypath,body
-
+    return api.cli_not_implemented(func)
 
 def run(func, args):
-    c = openconfig_lldp_client.Configuration()
-    c.verify_ssl = False
-    aa = openconfig_lldp_client.OpenconfigLldpApi(api_client=openconfig_lldp_client.ApiClient(configuration=c))
-
-    # create a body block
-    keypath, body = generate_body(func, args)
+    response = invoke_api(func, args)
+    if response.ok():
+        if response.content is not None:
+            # Get Command Output
+            api_response = response.content
 
-    try:
-        if body is not None:
-           api_response = getattr(aa,func.__name__)(*keypath, body=body)
-        else :
-           api_response = getattr(aa,func.__name__)(*keypath)
-        if api_response is None:
-            print ("Success")
-        else:
-            response = api_response.to_dict()
-            if 'openconfig_lldpinterfaces' in response.keys():
-                if not response['openconfig_lldpinterfaces']:
-                    return
-		neigh_list = response['openconfig_lldpinterfaces']['interface']
-                if neigh_list is None:
-                    return
-		show_cli_output(sys.argv[2],neigh_list)
-            elif 'openconfig_lldpinterface' in response.keys():
-	        neigh = response['openconfig_lldpinterface']#[0]['neighbors']['neighbor']
-                if neigh is None:
-                    return
-		if sys.argv[3] is not None:
-		    if neigh[0]['neighbors']['neighbor'][0]['state'] is None:
-			print('No LLDP neighbor interface')
-		    else:
-			show_cli_output(sys.argv[2],neigh)
-		else:
-	     	    show_cli_output(sys.argv[2],neigh)
+            if api_response is None:
+                print("api_response is None")
```

Comment:
Please do cleanup messages printed on the CLI.

Reply:
Done